src/node/desktop/src/main/detect-rosetta.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * detect-rosetta.ts
+ *
+ * Copyright (C) 2023 by Posit Software, PBC
+ *
+ * Unless you have received this program directly from Posit Software pursuant
+ * to the terms of a commercial license agreement with Posit Software, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+import { execSync } from 'child_process';
+import { logger } from '../core/logger';
+import { dialog, MessageBoxOptions } from 'electron';
+import { appState } from './app-state';
+
+enum DetectRosettaStatus {
+  ROSETTA_CHECK_FAILED,
+  ROSETTA_INSTALL_WARNING,
+}
+
+const DETECT_ROSETTA_STATUS_MAP: Map<DetectRosettaStatus, MessageBoxOptions> = new Map([
+  [
+    DetectRosettaStatus.ROSETTA_CHECK_FAILED,
+    {
+      type: 'error',
+      buttons: ['OK'],
+      defaultId: 0,
+      title: 'Rosetta Check Failed',
+      message: 'Something went wrong while detecting your Rosetta installation.',
+      detail: 'Please reopen RStudio to try again.',
+    },
+  ],
+  [
+    DetectRosettaStatus.ROSETTA_INSTALL_WARNING,
+    {
+      type: 'warning',
+      buttons: ['OK'],
+      defaultId: 0,
+      title: 'Rosetta Is Not Installed',
+      message: 'Rosetta is not installed on your machine. You may experience issues running RStudio.',
+      detail:
+        'To avoid these issues, please install Rosetta and reopen RStudio.\n\nFor instructions, please visit https://docs.posit.co/ide/desktop-pro/getting_started/installation.html#apple-silicon-mac-m1m2.',
```

Comment:
If I understand correctly, one downside to using a desktop dialog here is that users won't be able to copy + paste the URL here, or click on it to navigate.

What do you think about adding a button like "More Information" that, on click, would navigate to the above page?

Alternatively, we could use a GWT web dialog, where we'd have more control over the content of the dialog (and so could display clickable links).

Reply:
Users will be able to copy+paste, but since it's a pretty long URL, it's not great to have to select all that text, then copy+paste into a browser. I like the "More Information" button idea!

I've modified the buttons to now be: `More Information`, `Don't Remind Me Again` and `Remind Me Later`. I updated the screenshot in the description with the new buttons.